universal_transfer_operator/transfers/data_providers/base.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+from abc import ABC
+from typing import Any
+
+from airflow.hooks.base import BaseHook
+from datasets.base import UniversalDataset as Dataset
+
+from constants import FileLocation, LoadExistStrategy
+from utils import get_dataset_connection_type
+
+
+class DataProviders(ABC):
+    """
+    Base class to represent all the DataProviders interactions with Dataset.
+
+    The goal is to be able to support new dataset by adding
+    a new module to the `uto/data_providers` directory, without the need of
+    changing other modules and classes.
+    """
+
+    def __init__(
+        self,
+        dataset: Dataset,
+        transfer_mode,
+        transfer_params: dict = {},
+        if_exists: LoadExistStrategy = "replace",
```

Comment:
`if_exists` should be part of `transfer_params`

Reply:
https://github.com/astronomer/astro-sdk/pull/1492/commits/22b0aeeb515b6e9eb72dd68f62a64cb807a5732a